_pytest/main.py

```diff
@@ -520,8 +520,18 @@ def __init__(self, fspath, parent=None, config=None, session=None):
         super(FSCollector, self).__init__(name, parent, config, session)
         self.fspath = fspath
 
+    def _check_initialpaths_for_relpath(self):
+        for initialpath in self.session._initialpaths:
+            parent_path = self.fspath
+            for _ in parent_path.parts():
```

Comment:
This looks just wrong

Reply:
Yep, that's a fair comment. I've updated this to (hopefully) use the py.path apis properly.